src/components/user-settings-tabs/ProfileSettings.js

```diff
@@ -4,48 +4,134 @@ import { RaisedButton, TextField, } from 'material-ui'
 import PropTypes from 'prop-types'
 
 import BasicSpinner from '../BasicSpinner'
+import Dialog from './../../components/Dialog'
 import Theme from '../../../style/theme'
+import { validateEmail } from './../../utils/validator'
 
 const styles = {
   container: {
     padding: Theme.padding.tiny
+  },
+  content: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'flex-start'
+  },
+  button: {
+    marginTop: '20px'
   }
 }
 
 class ProfileSettings extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      dialogOpen: false,
+      email: props.user.email || '',
+      firstName: props.user.firstName || '',
+      lastName: props.user.lastName || '',
+      emailErrorText: ''
+    }
+
+    this.onSaveProfileSettings = this.onSaveProfileSettings.bind(this)
+    this.emailOnChange = this.emailOnChange.bind(this)
+    this.firstNameOnChange = this.firstNameOnChange.bind(this)
+    this.lastNameOnChange = this.lastNameOnChange.bind(this)
+    this.dialogClose = this.dialogClose.bind(this)
+  }
+
+  async onSaveProfileSettings() {
+    const { email, firstName, lastName } = this.state
+    const { user } = this.props
+    const validEmail = validateEmail(email)
+    if (validEmail) {
```

Comment:
The `else` block will never be reached. You need to replace this conditional with: `if (validEmail === true) {`

Reply:
TBH it might be worth re-writing the `validEmail` function. Maybe have it return an object: `{ success: boolean, errorMessage: String }` instead of a mixed return type that either returns a string or true. Because `true` and a non-empty string are both truthy values.